src/sealed.rs

```diff
@@ -0,0 +1,31 @@
+use crate::types::{
+    PyBool, PyByteArray, PyBytes, PyCapsule, PyComplex, PyDict, PyFloat, PyFrozenSet, PyList,
+    PyMapping, PyModule, PySequence, PySet, PySlice, PyString, PyTraceback, PyTuple, PyType,
+};
+use crate::{ffi, Bound, PyAny};
+
+pub trait Sealed {}
```

Comment:
Does it make sense to have a single `Sealed` trait for the whole crate, or should we have multiple `Sealed` traits each only relevant for one or more types?

We have `Sealed` for `PyResultExt` trait too, which could be merged in I guess, if it makes sense to do so.

Reply:
I can't think of an advantage of using multiple `Sealed` traits. The only relevant property is that it is a public trait, but not publicly nameable. If you can think of a good reason to split, I'm happy to do so, otherwise a central location is easy for reuse and might be easier to track changes in the future.

Hmm, now that I wrote that, maybe one disadvantage is that the `sealed` module is not completely private anymore, so implementation of the `Sealed` trait could be added anywhere in the crate instead of only in that module... How do you feel about that?

Reply:
I think I'm not worried about us implementing from anywhere in the crate, we can prevent that in review.

I think what I was wondering is, e.g. imagine we have a trait `Foo` which we only ever want implemented for `Bound<T>`. We seal it and we implement sealed for `Bound<T>`.

By adding that sealing for generic `T` now users can implement `PyDictMethods` for their own `Bound<My class>`.

If the sealing was done by separate traits then we could guarantee that the `PyDictMethods` seal would only ever allow `Bound<PyDict>` and not accidentally gave this mistake.

Reply:
Maybe I am putting the cart before the horse, and for now we should just live with one Sealed trait because orphan rules will stop users adding further implementations to any of the types we've sealed here.

If there's a need to seal for a generic, we can think carefully about that at the time 👍

Reply:
Ah yes, sealing a generic type will require more (careful) thoughts. I agree that we can think about that, once we have a use case for it.